packages/ember-data/lib/system/store.js

```diff
@@ -418,91 +418,24 @@ Store = Service.extend({
-    this method is the model's name as a string.
-
-    ---
-
-    To find a record by ID, pass the `id` as the second parameter:
-
-    ```javascript
-    store.find('person', 1);
-    ```
-
-    The `find` method will always return a **promise** that will be resolved
-    with the record. If the record was already in the store, the promise will
-    be resolved immediately. Otherwise, the store will ask the adapter's `find`
-    method to find the necessary data.
-
-    The `find` method will always resolve its promise with the same object for
-    a given type and `id`.
-
-    ---
-
-    You can optionally `preload` specific attributes and relationships that you know of
-    by passing them as the third argument to find.
-
-    For example, if your Ember route looks like `/posts/1/comments/2` and your API route
-    for the comment also looks like `/posts/1/comments/2` if you want to fetch the comment
-    without fetching the post you can pass in the post to the `find` call:
-
-    ```javascript
-    store.find('comment', 2, { preload: { post: 1 } });
-    ```
-
-    If you have access to the post model you can also pass the model itself:
-
-    ```javascript
-    store.find('post', 1).then(function (myPostModel) {
-      store.find('comment', 2, {post: myPostModel});
-    });
-    ```
-
-    This way, your adapter's `find` or `buildURL` method will be able to look up the
-    relationship on the record and construct the nested URL without having to first
-    fetch the post.
-
-    ---
-
-    To find all records for a type, call `findAll`:
-
-    ```javascript
-    store.findAll('person');
-    ```
-
-    This will ask the adapter's `findAll` method to find the records for the
-    given type, and return a promise that will be resolved once the server
-    returns the values. The promise will resolve into all records of this type
-    present in the store, even if the server only returns a subset of them.
-
     @method find
     @param {String} modelName
-    @param {(Object|String|Integer|null)} id
-    @param {Object} options
+    @param {String|Integer} id
     @return {Promise} promise
+    @private
   */
-  find: function(modelName, id, options) {
-    Ember.assert("You need to pass a type to the store's find method", arguments.length >= 1);
-    Ember.assert("You may not pass `" + id + "` as id to the store's find method", arguments.length === 1 || !Ember.isNone(id));
-    Ember.assert('Passing classes to store methods has been removed. Please pass a dasherized string instead of '+ Ember.inspect(modelName), typeof modelName === 'string');
-
-    if (arguments.length === 1) {
-      Ember.assert('Using store.find(type) has been removed. Use store.findAll(type) to retrieve all records for a given type.');
-    }
-
-    // We are passed a query instead of an id.
-    if (Ember.typeOf(id) === 'object') {
-      Ember.assert('Calling store.find() with a query object is no longer supported. Use store.query() instead.');
-    }
```

Comment:
I like the idea of keeping asserts for the old `store.find` usages so people copy and pasting old code get a helpful error message.